GoogleForCreators/web-stories-wp
# Story Details Modal: Publish + Checklist Button Parity

## Context

Parity that was left out of initial plan for the story details modal in 1.18 (edge case clean up)

## Summary

The prepublish checklist has checkpoints that decide when to expose sections of the checklist. The one that got out of sync is for priority issues - which exposed at 4th page add _or_ when 'publish' in header is clicked - until then no badges will show up. Basically, we should be using the same "publish" button for both instances of it so that the logic for what version of it stay parallel. 

While doing that, noticed we could improve the flow for contributors - since that was tied in with the publish button I was already working on I did it here as well because it would have been more work to do it separately. Now, if you're a contributor (local dev env we have this set up already `contributor` `password`) and you click 'submit for review' you also get the story details modal with contributor approved options to finalize before you submit. 

## Relevant Technical Choices

- new action in checkpoint context to independently update to expose priority issues. 
- Consolidate `buttonWithChecklistWarning` to remove redundancies between the two publish buttons since they should behave the same - so now most of that logic lives there instead of in the header or the dialog/getting passed in from the header. 
- Removing check for canPublish on header publish button because regardless of that ability, users should all see this story details modal and then proceed from there accordingly.

## To-do

- [x] New e2e test for contributor 

It might be nice for a dialog to pop up for contributors telling them their story was submitted for review. Drew the line on that for scope creep here :) 

Also looking forward to cleaning up the checklist provider once story details experiment is deleted next sprint.


## User-facing changes

### Checklist Parity
- The checklist toggle button on the footer and the checklist button on the story details modal should have the same badges at all times now - when one changes, so does the other 
- The publish button in the header and the publish button in the story details modal should have the same copy, tooltips, badges at all times - when one changes, so does the other 

| Checklist Parity Demos | |
| --- | --- |
| https://user-images.githubusercontent.com/10720454/156441165-edabcd56-9a46-491d-9ef9-307b6e169ed5.mp4 | https://user-images.githubusercontent.com/10720454/156441220-57e020e6-0c76-4b85-8cc3-96b747c1fd40.mp4 |

### Contributor Flow 
- Now when the 'submit for review' button is hit in the header, contributor also gets story details modal. Here they can finalize before submitting and then when they hit submit in the modal it'll have the same flow as before (now the modal closes) and header updates like usual.

| Existing contributor flow | New contributor flow  |
| --- | --- |
| https://user-images.githubusercontent.com/10720454/156441360-1074b011-0665-4d32-9227-35fdbb9cd965.mp4 | https://user-images.githubusercontent.com/10720454/156441419-4856cd2a-1d40-4222-839b-bce1b8aa6568.mp4 |


## Testing Instructions

See the above 😄  

- Make sure publish and checklist buttons stay in sync in editor and modal 
- Make sure contributor can still submit stories for review 




This PR can be tested by following these steps:

1.


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #10623
